File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Lucky_PIN.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N;
string S;

string i2s(ll x) {
  string res = to_string(x);
  ll n = res.size();
  for (int i = 0; n + i < 3; ++i) {
    res = '0' + res;
  }
  return res;
}

void solve() {
  ll res = 0;
  for (int i = 0; i < 1000; ++i) {
    string n = i2s(i);
    int idx = 0;
    for (auto s : S) {
      if (s == n[idx]) {
        ++idx;
        if (idx >= 3) {
          break;
        }
      }
    }
    if (idx >= 3) {
      ++res;
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N >> S;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/D_Lucky_PIN.py
N = int(input())
S = input()

res = 0
for i in range(1000):
    n = f"{i:03d}"
    idx = 0
    for s in S:
        if s == n[idx]:
            idx += 1
            if idx >= 3:
                break
    if idx >= 3:
        res += 1
print(res)
